File: lib/models/resource.dart
class Resource {
  final int id;
  final String name;
  final String category;
  final String department;
  final String location;
  final String description;
  final int quantity;
  final int availableQuantity;
  final double utilization;
  final String status;
  final DateTime? createdAt;

  const Resource({
    required this.id,
    required this.name,
    required this.category,
    required this.department,
    required this.location,
    required this.description,
    required this.quantity,
    required this.availableQuantity,
    required this.utilization,
    required this.status,
    this.createdAt,
  });

  factory Resource.fromJson(Map<String, dynamic> json) {
    return Resource(
      id: json['id'] as int,
      name: json['name'] as String,
      category: json['category'] as String,
      department: json['department'] as String,
      location: json['location'] as String,
      description: json['description'] as String,
      quantity: json['quantity'] as int,
      availableQuantity: json['available_quantity'] as int,
      utilization: (json['utilization'] as num).toDouble(),
      status: json['status'] as String,
      createdAt: json['created_at'] != null
          ? DateTime.tryParse(json['created_at'].toString())
          : null,
    );
  }

  bool get isAvailable => availableQuantity > 0;

  bool get isLimited => status.toLowerCase() == 'limited';
}

File: backend/app/models/resource.py
from datetime import datetime

from sqlalchemy import DateTime, Float, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from ..database import Base


class Resource(Base):
    __tablename__ = "resources"

    id: Mapped[int] = mapped_column(
        primary_key=True,
        index=True,
    )

    name: Mapped[str] = mapped_column(
        String(150),
        nullable=False,
    )

    category: Mapped[str] = mapped_column(
        String(100),
        nullable=False,
    )

    department: Mapped[str] = mapped_column(
        String(120),
        nullable=False,
    )

    location: Mapped[str] = mapped_column(
        String(150),
        nullable=False,
    )

    description: Mapped[str] = mapped_column(
        Text,
        default="",
        nullable=False,
    )

    quantity: Mapped[int] = mapped_column(
        Integer,
        default=1,
        nullable=False,
    )

    available_quantity: Mapped[int] = mapped_column(
        Integer,
        default=1,
        nullable=False,
    )

    utilization: Mapped[float] = mapped_column(
        Float,
        default=0.0,
        nullable=False,
    )

    status: Mapped[str] = mapped_column(
        String(30),
        default="Available",
        nullable=False,
    )

    created_at: Mapped[datetime] = mapped_column(
        DateTime,
        default=datetime.utcnow,
        nullable=False,
    )